packages/office-ui-fabric-react/src/components/DetailsList/ShimmeredDetailsList.types.ts

```diff
@@ -1,24 +1,78 @@
 import { IDetailsListProps } from './DetailsList.types';
-import { IStyle, ITheme } from '../../Styling';
+import { IDetailsRowProps } from './DetailsRow.types';
+import { IStyle } from '../../Styling';
 import { IStyleFunctionOrObject } from '../../Utilities';
 
+/**
+ * ShimmeredDetailsList props interface
+ * {@docCategory ShimmeredDetailsList}
+ */
 export interface IShimmeredDetailsListProps extends IDetailsListProps {
-  /** The theme provided by context */
-  theme?: ITheme;
+  /**
+   * DetailsList `styles` to pass through.
+   * Due to ShimmeredDetailsList overriding the extended `styles` prop with its own one
+   * we temporary introduce this one slot to allow `DetailsList` styles customization.
+   * @deprecated Will be removed in Fabric 7.0 in favor of `styles` prop that will have its typing adjusted
+   */
+  detailsListStyles?: IDetailsListProps['styles'];
 
-  /** Overridable styles */
+  /**
+   * Boolean flag to control when to render placeholders vs real items.
+   * It's up to the consumer app to know when fetching of the data is done to toggle this prop.
+   */
+  enableShimmer?: boolean;
+
+  /**
+   * Determines whether to remove a fading out to bottom overlay over the shimmering items
+   * used to further emphasize the unknown number of items that will be fetched.
+   */
+  removeFadingOverlay?: boolean;
+
+  /**
+   * Custom placeholder renderer to be used when in need to override the default placeholder of a DetailsRow.
+   * Optional argument is passed to leverage the calculated column measurements done by DetailsList.
+   */
+  onRenderCustomPlaceholder?: (rowProps?: IDetailsRowProps) => React.ReactNode;
```

Comment:
Is `rowProps` _optional_ or is it always provided? Seems like its provided given the above call.

https://github.com/OfficeDev/office-ui-fabric-react/pull/8744/files#diff-2d7d3d7e7065bb4674f1f9ed75572ab2R73

Reply:
Wait... I am not following your question. It should be optional and be used by users only if they want to leverage them when they define the callback. Am I missing something?

Reply:
The callback should be marked as `optional` but the `rowProps` argument seems to be always provided as in the LoC linked above. So the argument should probably be made non-optional, unless there's a case where `undefined` is passed-in